app/security_system.py

```diff
@@ -66,13 +67,36 @@ def recognize_face(self, frame):
 
         # TODO 3
         # Perform face detection
+        faces = face_cascade.detectMultiScale(gray_frame, scaleFactor = 1.2, minNeighbors = 5, minSize = (30,30))
         # Iterate through each detected face
-        # Extract the detected face region
-        # Perform facial recognition on the detected face
-        # Draw bounding box around the detected face
-        # Annotate the frame with the recognized user's name and authorization status
-        # Log access attempt
-        # Your code goes here
+        for face in faces:
+            # Extract the detected face region
+            x,y,w,h = face
+            X,Y = x+w,y+h
+            image = gray_frame[x:X, y:Y]
+            # Perform facial recognition on the detected face
+            label, confidence = self.face_recognizer.predict(image)
+            # Draw bounding box around the detected face
+            l= 20
+            t= 5
+            cv2.rectangle(frame,face,(255,0,200),1)
+            cv2.line(frame,(x,y),(x+l,y),(255,0,200),t)
+            cv2.line(frame,(x,y),(x,y+l),(255,0,200),t)
+            cv2.line(frame,(X-l,y),(X,y),(255,0,200),t)
+            cv2.line(frame,(X,y),(X,y+l),(255,0,200),t)
+            cv2.line(frame,(x,Y-l),(x,Y),(255,0,200),t)
+            cv2.line(frame,(x,Y),(x+l,Y),(255,0,200),t)
+            cv2.line(frame,(X-l,Y),(X,Y),(255,0,200),t)
+            cv2.line(frame,(X,Y-l),(X,Y),(255,0,200),t)
+            # Annotate the frame with the recognized user's name and authorization status
+            name = self.get_person_name(label,confidence)
+            authorization_status = self.authorized_persons[name]
```

Comment:
The `authorization_status` will hold values `true` or `false`. So, it makes more sense to put the name of variable as something like `is_authorized`.